assessments-service/src/main/java/com/transformuk/hee/tis/assessment/service/service/impl/AssessmentServiceImpl.java

```diff
@@ -286,6 +286,49 @@ public boolean deleteTraineeAssessment(Long assessmentId, Long traineeId) {
     return false;
   }
 
+  private AssessmentDTO updateAssessmentWithNestedDtos(AssessmentDTO assessmentDto) {
+    Assessment assessment = assessmentMapper.toEntity(assessmentDto);
+    AssessmentDetailDTO assessmentDetailDto = null;
+    AssessmentOutcomeDTO assessmentOutcomeDto = null;
+    RevalidationDTO revalidationDto = null;
+
+    if (assessmentDto.getDetail() != null) {
+      if (assessmentDto.getDetail().getId() != null) {
+        assessmentDetailDto = assessmentDetailService.save(assessment,
+            assessmentDto.getDetail());
+      } else {
+        assessmentDetailDto = assessmentDetailService.create(assessment,
+            assessmentDto.getDetail());
```

Comment:
The id will be set in `assessmentDetailService.create` using `assessment.id`: https://github.com/Health-Education-England/TIS-ASSESSMENTS/blob/06ccebb314a192c792934e4913e5aa7cbb9dde5d/assessments-service/src/main/java/com/transformuk/hee/tis/assessment/service/service/impl/AssessmentDetailServiceImpl.java#L88